src/decoder/stream.rs

```diff
@@ -1229,6 +1229,15 @@ impl StreamingDecoder {
             }
         };
 
+        if let Some(mut raw_row_len) = color_type.checked_raw_row_length(bit_depth, width) {
+            if interlaced {
+                // This overshoots, but overestimating should be fine.
+                raw_row_len = raw_row_len.saturating_mul(2);
```

Comment:
Does this look okay?

Reading https://en.wikipedia.org/wiki/Adam7_algorithm, I see that:

* Pass 1 fills 1/64 of pixels
* Pass 2 fills additional 1/64 of pixels (for a total of 1/32)
* Pass 3 fills additional 1/32 of pixels
* Pass 4 fills additional 1/16 of pixels
* Pass 5 fills additional 1/8 of pixels
* Pass 6 fills additional 1/4 of pixels
* Pass 7 fills additional 1/2 of pixels
 
And this unsurprisingly adds up to 100% of pixels.  But I am not sure if there may be a need for some extra data (more "filter" type bytes;  maybe pixel bits need to be rounded up to the nearest byte), so I thought that mutiplying by 2 would be safest.

Reply:
Would be nice to precisely figure out how many bytes we need, but I'm OK with leaving this as a TODO for now